core/trino-main/src/main/java/io/trino/sql/analyzer/StatementAnalyzer.java

```diff
@@ -807,7 +815,87 @@ protected Scope visitAnalyze(Analyze node, Optional<Scope> scope)
                 throw new AccessDeniedException(format("Cannot ANALYZE (missing insert privilege) table %s", tableName));
             }
 
-            analysis.setAnalyzeTarget(tableHandle);
+            analysis.setAnalyzeTargets(ImmutableList.of(tableHandle));
+            return createAndAssignScope(node, scope, Field.newUnqualified("rows", BIGINT));
+        }
+
+        @Override
+        protected Scope visitAnalyzeForQuery(AnalyzeForQuery node, Optional<Scope> scope)
+        {
+            analysis.setUpdateType("ANALYZE");
+
+            checkState(queryAnalyzer.isPresent(), "Query analyzer must be provided for ANALYZE FOR QUERY");
+
+            Query query = node.getQuery();
+
+            // analyze statement
+            Analysis queryAnalysis = queryAnalyzer.get().analyze(session, query, ImmutableList.copyOf(analysis.getParameters().values()), warningCollector);
+
+            ImmutableSet.Builder<QualifiedObjectName> tableNameBuilder = ImmutableSet.builder();
+            ImmutableSet.Builder<QualifiedObjectName> viewNameBuilder = ImmutableSet.builder();
+            queryAnalysis.getTableNodeRefs().stream()
+                    .map(NodeRef::getNode)
+                    .forEach(table -> {
+                        QualifiedObjectName tableName = createQualifiedObjectName(session, table, table.getName());
+                        if (metadata.getView(session, tableName).isPresent()) {
+                            viewNameBuilder.add(tableName);
+                        }
+                        else {
+                            tableNameBuilder.add(tableName);
+                        }
+                    });
+            Set<QualifiedObjectName> viewNames = viewNameBuilder.build();
+            Set<QualifiedObjectName> tableNames = tableNameBuilder.build();
+
+            // verify the target tables exists, and it's not all views
+            if (tableNames.isEmpty()) {
+                throw semanticException(NOT_SUPPORTED, node, "Analyzing views is not supported");
```

Comment:
I think this situation also occurs if you run a query like
```
ANALYZE FOR SELECT 123
```
So this error should probably be something like "No tables to analyze"